### ISSUE TITLE
reflect: CL 20968 breaks reflect test on FreeBSD/ARM

### ISSUE THREAD

**Issue by paulzhol** [golang/go#14962]
SIGBUS is raised from a pointer dereference in runtime/type.go:282.

http://build.golang.org/log/2ec4336285d65cc33267f147a2f7b50631eefbe5
```
fatal error: unexpected signal during runtime execution
[signal 0xa code=0x0 addr=0x1e21e7 pc=0x62b00]

goroutine 65 [running]:
runtime.throw(0x1f6052, 0x2a)
	/home/yuval/buildroot/freebsd-arm-paulzhol-24ce64d1a967/go/src/runtime/panic.go:566 +0x78 fp=0x3075e5e8 sp=0x3075e5dc
runtime.sigpanic()
	/home/yuval/buildroot/freebsd-arm-paulzhol-24ce64d1a967/go/src/runtime/sigpanic_unix.go:12 +0x44 fp=0x3075e614 sp=0x3075e5e8
runtime.(*name).pkgPath(0x1c1c00, 0x1e21e6)
	/home/yuval/buildroot/freebsd-arm-paulzhol-24ce64d1a967/go/src/runtime/type.go:282 +0x108 fp=0x3075e628 sp=0x3075e618
runtime.getitab(0x1c1bd0, 0x1c7ee0, 0x8e300, 0x2012f)
	/home/yuval/buildroot/freebsd-arm-paulzhol-24ce64d1a967/go/src/runtime/iface.go:89 +0x354 fp=0x3075e684 sp=0x3075e628
runtime.typ2Itab(0x1c7ee0, 0x1c1bd0, 0x66358c, 0x3b9aca00)
	/home/yuval/buildroot/freebsd-arm-paulzhol-24ce64d1a967/go/src/runtime/iface.go:132 +0x30 fp=0x3075e69c sp=0x3075e684
reflect_test.TestUnexportedMethods(0x30952880)
	/home/yuval/buildroot/freebsd-arm-paulzhol-24ce64d1a967/go/src/reflect/all_test.go:2378 +0x60 fp=0x3075e7b0 sp=0x3075e69c
testing.tRunner(0x30952880, 0x5c8dc4)
	/home/yuval/buildroot/freebsd-arm-paulzhol-24ce64d1a967/go/src/testing/testing.go:547 +0xa4 fp=0x3075e7d4 sp=0x3075e7b0
runtime.goexit()
	/home/yuval/buildroot/freebsd-arm-paulzhol-24ce64d1a967/go/src/runtime/asm_arm.s:993 +0x4 fp=0x3075e7d4 sp=0x3075e7d4
created by testing.(*T).Run
	/home/yuval/buildroot/freebsd-arm-paulzhol-24ce64d1a967/go/src/testing/testing.go:578 +0x320

goroutine 1 [chan receive]:
testing.(*T).Run(0x30952880, 0x1ec4c7, 0x15, 0x5c8dc4, 0x14ab01)
	/home/yuval/buildroot/freebsd-arm-paulzhol-24ce64d1a967/go/src/testing/testing.go:579 +0x340
testing.RunTests.func1(0x307b6080)
	/home/yuval/buildroot/freebsd-arm-paulzhol-24ce64d1a967/go/src/testing/testing.go:724 +0xb0
testing.tRunner(0x307b6080, 0x307afec0)
	/home/yuval/buildroot/freebsd-arm-paulzhol-24ce64d1a967/go/src/testing/testing.go:547 +0xa4
testing.RunTests(0x5c8ad4, 0x653c18, 0x74, 0x74, 0x1)
	/home/yuval/buildroot/freebsd-arm-paulzhol-24ce64d1a967/go/src/testing/testing.go:730 +0x428
testing.(*M).Run(0x307aff6c, 0x0)
	/home/yuval/buildroot/freebsd-arm-paulzhol-24ce64d1a967/go/src/testing/testing.go:675 +0x8c
main.main()
	reflect/_test/_testmain.go:304 +0x118

goroutine 38 [sleep]:
time.Sleep(0x3b9aca00, 0x0)
	/home/yuval/buildroot/freebsd-arm-paulzhol-24ce64d1a967/go/src/runtime/time.go:59 +0xec
reflect_test.selectWatcher()
	/home/yuval/buildroot/freebsd-arm-paulzhol-24ce64d1a967/go/src/reflect/all_test.go:1392 +0x24
created by reflect_test.TestSelect.func1
	/home/yuval/buildroot/freebsd-arm-paulzhol-24ce64d1a967/go/src/reflect/all_test.go:1122 +0x24
FAIL	reflect	0.271s
```

**Comment by ianlancetaylor** [comment]
@crawshaw The alignment you are doing for the pkgpath pointer assumes that the name.bytes field itself points to aligned memory, but nothing ensures that.  Sorry for not catching that.

**Comment by crawshaw** [comment]
I fixed this by accident in 4d92041. The linker aligns the start of most symbols, but not go.string.* symbols. With 4d92041, the type data is written into an aligned symbol.

Do you think that's enough here, or should I modify the code to defend against future unaligned symbols? I don't see it being very likely, in general the .type.* symbols are full of pointers so it's not safe for the linker to pack them. And there aren't very many of these.

**Comment by ianlancetaylor** [comment]
The change sounds sufficient.  But since most people run x86, and since x86 permits unaligned memory loads, I think it would be a good idea to a test to the reflect or runtime package that the final address of the pointer is in fact aligned.  Or maybe just panic in reflect if the pointer is not aligned.
